src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java

```diff
@@ -0,0 +1,83 @@
+package com.based.basedsurvey.controller;
+
+import com.based.basedsurvey.data.MultiplechoiceQuestion;
+import com.based.basedsurvey.data.Question;
+import com.based.basedsurvey.data.QuestionTypes;
+import com.based.basedsurvey.data.Survey;
+import com.based.basedsurvey.repo.QuestionRepository;
+import com.based.basedsurvey.repo.SurveyRepository;
+import lombok.extern.java.Log;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Log
+@Controller
+public class WebEditSurveyController {
+
+    SurveyRepository surveyRepository;
+    QuestionRepository questionRepository;
+    @Autowired
+    public WebEditSurveyController(SurveyRepository surveyRepository, QuestionRepository questionRepository){
+        this.surveyRepository = surveyRepository;
+        this.questionRepository = questionRepository;
+
+    }
+
+    @GetMapping(path = "survey/{surveyId}/edit")
+    public String editSurvey(@PathVariable("surveyId") long id, Model model) {
+        var survey = getSurvey(id);
+        model.addAttribute("questions", questionRepository.findAllBySurveyId(survey.getId()));
+        model.addAttribute("survey", survey);
+        return "newSurvey";
+    }
+
+    @PostMapping("question/create")
+    public String createQuestion(@RequestParam long surveyID, @RequestParam QuestionTypes qt, @RequestParam String prompt) {
+        var question = QuestionTypes.makeQuestionFromType(qt);
+        var survey = getSurvey(surveyID);
+        question.setSurvey(survey);
+        question.setPrompt(prompt);
+        questionRepository.save(question);
+        survey.getQuestions().add(question);
+        surveyRepository.save(survey);
+        return "redirect:/survey/"+surveyID +"/edit";
+    }
+
+    @PostMapping("question/delete")
+    public String deleteQuestion(@RequestParam long questionID, @RequestParam long surveyID) {
+        var survey = getSurvey(surveyID);
+        var question = getQuestion(questionID);
+        survey.getQuestions().remove(question);
+        surveyRepository.save(survey);
+        return "redirect:/survey/"+surveyID +"/edit";
+    }
+
+    @PostMapping("survey/rename")
+    public String editName(@RequestParam String prompt, @RequestParam long id) {
+        var survey = getSurvey(id);
+        survey.setName(prompt);
+        surveyRepository.save(survey);
+        return "redirect:/survey/"+id +"/edit";
+    }
+
```

Comment:
getSurvey and getQuestion are also defined in WebQuestionController, maybe make some helper class that has these methods.

Reply:
this can be done for milestone 2 #33